shell/server/hack-session.js

```diff
@@ -270,12 +276,22 @@ Meteor.methods({
       created: new Date(),
       expires: null,
       forSharing: forSharing,
+      expiresIfUnused: selfDestructAt
     });
 
     return {token: token, endpointUrl: endpointUrl};
   }
 });
 
+var cleanupSelfDestructing = function() {
+  var now = new Date();
+  ApiTokens.remove({expiresIfUnused: {$lt: now}});
+};
+
+// Make self-destructing tokens actually self-destruct, so they don't
```

Comment:
Could we also not display the token in the webkey UI until it has had its first use?